include/pstl_offload/internal/usm_memory_replacement.h

```diff
@@ -53,34 +53,58 @@ __get_offload_device()
 class __offload_policy_holder_type
 {
     using __set_active_device_func_type = void (*)(sycl::device*);
+    using __get_offload_device_func_type = sycl::device (*)();
 
   public:
     // Since the global object of __offload_policy_holder_type is static but the constructor
     // of the class is inline, we need to avoid calling static functions inside of the constructor
     // and pass the pointer to exact function as an argument to guarantee that the correct __active_device
     // would be stored in each translation unit
-    __offload_policy_holder_type(sycl::device&& __offload_device,
+    __offload_policy_holder_type(__get_offload_device_func_type __get_offload_device_func,
```

Comment:
May be it would be better to change `get_offload_device` to return device selector and pass the selector into the constructor?

Reply:
Done.